base_layer/core/src/base_node/state_machine_service/state_machine.rs

```diff
@@ -215,6 +223,39 @@ impl<B: BlockchainBackend + 'static> BaseNodeStateMachine<B> {
         let _ = self.service_shutdown.trigger();
     }
 
+    // Function to handle the recovery attempt of the db
+    fn recover_db<D: BlockchainBackend + 'static>(self, temp_db: D) -> Result<(), String> {
```

Comment:
I don't see why the recover_db function has to be part of the base node state machine.

Can this not all happen in the base_node_application? All you need to db access.

Reply:
You need the backend to run as normal as it needs to do all the validation that would normally run on a new block. 
By starting up the state_machine, we know the internal backend is run normally, and we have full control over the state, and can ensure that the statemachine does not enter a sync state.

Reply:
> You need the backend to run as normal as it needs to do all the validation that would normally run on a new block.

This happens independently of the state machine so I don't see this as a good argument for putting this code here.

> By starting up the state_machine, we know the internal backend is run normally, and we have full control over the state, and can ensure that the statemachine does not enter a sync state.

Do we ensure that the state machine does not enter a sync state? Do we need to? None of the code does what you are saying here.

Reply:
Agree with @sdbondi. Essentially, the idea (recovery) is a good one, but this implementation is a bit hacky that violates several SOLID principles. If it were refactored into a separate module that reused the existing mechanisms, I would approve it.